utils/hct/hctdb.py

```diff
@@ -5848,10 +5848,98 @@ def UFI(name, **mappings):
+              "Matrix-Vector Multiply",
+              "<hfwi,<hfwi",
+              "",
+              [
+                  db_dxil_param(0, "$x0", "outputVector", "output vector"),
+                  db_dxil_param(2, "$x1", "inputVector", "input vector"),
+                  db_dxil_param(3, "i1", "isInputUnsigned", "input signed op kind"),
+                  db_dxil_param(4, "i32", "inputInterpretation", "input interpretation"),
+                  db_dxil_param(5, "res", "matrixBuffer", "matrix resource"),
+                  db_dxil_param(6, "i32", "matrixOffset", "matrix offset"),
+                  db_dxil_param(7, "i32", "matrixIntepretation", "matrix intepretation"),
+                  db_dxil_param(8, "i32", "M", "matrix M dimension"),
+                  db_dxil_param(9, "i32", "K", "matrix K dimension"),
+                  db_dxil_param(10, "i32", "matrixLayout", "matrix layout"),
+                  db_dxil_param(11, "i1", "matrixTranspose", "matrix transpose"),
+                  db_dxil_param(12, "i32", "matrixStride", "matrix stride"),
+                  db_dxil_param(13, "i1", "isOutputUnsigned", "output signed op kind"),
+              ],
+          )
+        next_op_idx += 1
+
+        self.add_dxil_op(
+              "MatVecMulAdd",
+              next_op_idx,
+              "MatVecMulAdd",
+              "Matrix-Vector Multiply Add",
+              "<hfwi,<hfwi",
+              "",
+              [
+                  db_dxil_param(0, "$x0", "outputVector", "output vector"),
+                  db_dxil_param(2, "$x1", "inputVector", "input vector"),
+                  db_dxil_param(3, "i1", "isInputUnsigned", "input signed op kind"),
+                  db_dxil_param(4, "i32", "inputInterpretation", "input interpretation"),
+                  db_dxil_param(5, "res", "matrixBuffer", "matrix resource"),
+                  db_dxil_param(6, "i32", "matrixOffset", "matrix offset"),
+                  db_dxil_param(7, "i32", "matrixIntepretation", "matrix intepretation"),
+                  db_dxil_param(8, "i32", "M", "matrix M dimension"),
+                  db_dxil_param(9, "i32", "K", "matrix K dimension"),
+                  db_dxil_param(10, "i32", "matrixLayout", "matrix layout"),
+                  db_dxil_param(11, "i1", "matrixTranspose", "matrix transpose"),
+                  db_dxil_param(12, "i32", "matrixStride", "matrix stride"),
+                  db_dxil_param(13, "res", "biasBuffer", "bias vector resource"),
+                  db_dxil_param(14, "i32", "biasOffset", "bias vector offset"),
+                  db_dxil_param(15, "i32", "biasIntepretation", "bias vector intepretation"),
+                  db_dxil_param(16, "i1", "isOutputUnsigned", "output signed op kind"),
+              ],
+          )
+        next_op_idx += 1
+
+        self.add_dxil_op(
+              "OuterProductAccumulate",
+              next_op_idx,
+              "OuterProductAccumulate",
+              "Outer Product Accumulate",
```

Comment:
nit: It would be nice to have better one-line descriptions for these operations.